src/Lecture3.hs

```diff
@@ -137,10 +161,18 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
+  (<>) :: List1 a -> List1 a -> List1 a
+  (List1 x xs) <> (List1 y ys) = List1 x (xs <> (y:ys))
 
 
 {- | Does 'List1' have the 'Monoid' instance? If no then why?
 
+jakalx: There cannot be an instance for Monoid since we cannot construct an 'a' out
+of thin air without knowing anything about this a. Even if 'a' was a Monoid, we
+could not satisfy:
+
+  mempty <> a = a <> mempty
```

Comment:
That's correct! 👍🏻 
Or, in other words, there's no such thing as "empty non-empty list" 🙂